### PR TITLE
Remove the Social Override from Phoenix

### PR DESCRIPTION
**Pull request by mendelB**
### What's this PR do?

This pull request completely retires the legacy/deprecated Social Override field from Campaigns/Pages/Campaign Updates, and all associated code.

### How should this be reviewed?
Just a whole lotta 🗑 ! commit-by-commit could be convenient.
I did my best to confirm that nothing active is being deprecated, but if you spot something give a shout! 

### Any background context you want to provide?
The initiative to replace this field with the Metadata content type happened a whole while back starting around https://github.com/DoSomething/phoenix-next/pull/1137, this just purges the code relics!

I got [confirmation](https://dosomething.slack.com/archives/C09ANFQLA/p1595258188069000) that we're ok to retire the only existing (_non functional) instances of these in Campaign Updates so I'll remove those from the Contentful space.

### Relevant tickets

References [Pivotal #173771700](https://www.pivotaltracker.com/story/show/173771700).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added screenshots of front-end changes on small, medium, and large screens.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

**Comment by mendelB** [comment]
Ahhh flip. I forgot that I branched https://github.com/DoSomething/phoenix-next/pull/2286 off of this one, and merged it into here! If you commit-by-commit this one, just ignore the final commit with the additional Campaign Update changes!

File: contentful/content-types/campaign.js [hunk 1]
```diff
@@ -431,8 +418,8 @@ module.exports = function(migration) {
         nodes: {},
       },
       {
-        enabledMarks: ['bold', 'underline', 'italic'],
-        message: 'Only bold, underline, and italic marks are allowed',
+        enabledMarks: ['bold', 'italic', 'underline'],
+        message: 'Only bold, italic, and underline marks are allowed',
```
Comment (Dinnall) [inline contentful/content-types/campaign.js:422]: Curious to know why these were changed here

File: resources/assets/components/ContentfulEntry/ContentfulEntry.js [hunk 2]
```diff
@@ -80,7 +80,7 @@ class ContentfulEntry extends React.Component {
 
       case 'CampaignUpdateBlock':
         return (
-          <CampaignUpdateContainer
+          <CampaignUpdate
```
Comment (Dinnall) [inline resources/assets/components/ContentfulEntry/ContentfulEntry.js:83]: What do you think of the name `UpdateCampaign` here?

Comment (mendelB) [reply resources/assets/components/ContentfulEntry/ContentfulEntry.js:83]: We're rendering the [`CampaignUpdate` component](https://github.com/DoSomething/phoenix-next/blob/40c785c3cfa5ee507c51519bcc49fdffe9fde121/resources/assets/components/blocks/CampaignUpdate/CampaignUpdate.js) here so we need to invoke it as named. (This is one of our older blocks which allows Campaign Leads to embed an update / link / announcement within campaign pages via Contentful so we called it 'Campaign Update')

File: contentful/content-types/campaign.js (hunk 1 above)
Comment (mendelB) [reply contentful/content-types/campaign.js:422]: haha this drove me crazy bc I couldn't figure out why the heck the order kept on changing when I generated the migration from the `dev` & `master` spaces on Contentful! Something must be stored in a different order somehow which causes the discrepancy 🤷‍♂️

File: resources/assets/components/ContentfulEntry/ContentfulEntry.js (hunk 2 above)
Comment (Dinnall) [reply resources/assets/components/ContentfulEntry/ContentfulEntry.js:83]: ah- makes sense

**Pull request merged**